internal/cli/actions.go

```diff
@@ -0,0 +1,540 @@
+				Dependencies: apiActionDependenciesFor(inputs.Dependencies),
+				Secrets:      apiActionSecretsFor(inputs.Secrets),
+			}
+
+			if err := ansi.Waiting(func() error {
+				return cli.api.Action.Create(action)
+			}); err != nil {
+				return fmt.Errorf("An unexpected error occurred while attempting to create an action with name '%s': %w", inputs.Name, err)
+			}
+
+			cli.renderer.ActionCreate(action)
+			return nil
+		},
+	}
+
+	actionName.RegisterString(cmd, &inputs.Name, "")
+	actionTrigger.RegisterString(cmd, &inputs.Trigger, "")
+	actionCode.RegisterString(cmd, &inputs.Code, "")
+	actionDependency.RegisterStringMap(cmd, &inputs.Dependencies, nil)
+	actionSecret.RegisterStringMap(cmd, &inputs.Secrets, nil)
+
+	return cmd
+}
+
+func updateActionCmd(cli *cli) *cobra.Command {
+	var inputs struct {
+		ID string
+		Name         string
+		Trigger      string
+		Code         string
+		Dependencies map[string]string
+		Secrets      map[string]string
+	}
+
+	cmd := &cobra.Command{
+		Use:   "update",
+		Args:  cobra.MaximumNArgs(1),
+		Short: "Update an action",
+		Long:  "Update an action.",
+		Example: `auth0 actions update <id> 
+auth0 actions update <id> --name myaction
+auth0 actions update <id> --n myaction --trigger post-login
+auth0 actions update <id> --n myaction -t post-login -d "lodash=4.0.0" -d "uuid=8.0.0"
+auth0 actions update <id> --n myaction -t post-login -d "lodash=4.0.0" -s "API_KEY=value" -s "SECRET=value`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				inputs.ID = args[0]
+			} else {
+				err := actionID.Pick(cmd, &inputs.ID, cli.actionPickerOptions)
+				if err != nil {
+					return err
+				}
+			}
+
+			var current *management.Action
+			err := ansi.Waiting(func() error {
+				var err error
+				current, err = cli.api.Action.Read(inputs.ID)
+				return err
+			})
+			if err != nil {
+				return fmt.Errorf("Failed to fetch action with ID: %s %v", inputs.ID, err)
+			}
+
+			if err := actionName.AskU(cmd, &inputs.Name, current.Name); err != nil {
+				return err
+			}
+
+			err, triggers, version := latestActionTriggers(cli)
+			if err != nil {
+				return err
+			}
+
+			if err := actionTrigger.SelectU(cmd, &inputs.Trigger, triggers, nil); err != nil {
+				return err
+			}
+
+			// TODO(cyx): we can re-think this once we have
+			// `--stdin` based commands. For now we don't have
+			// those yet, so keeping this simple.
+			if err := actionCode.EditorPromptU(
+				cmd,
+				&inputs.Code,
+				actionTemplate(inputs.Trigger),
+				inputs.Name+".*.js",
+				cli.actionEditorHint,
+			); err != nil {
+				return err
+			}
+			if err != nil {
+				return fmt.Errorf("Failed to capture input from the editor: %w", err)
+			}
+
+			if inputs.Name == "" {
+				inputs.Name = current.GetName()
+			}
+
+			if inputs.Trigger == "" && len(current.SupportedTriggers) > 0 {
+				inputs.Trigger = current.SupportedTriggers[0].GetID()
+			}
+
+			unprexifedVersion := strings.TrimPrefix(version, "v")
```

Comment:
Actions update fails with the prefixed version that comes from the triggers endpoint (eg `v2`):
<img width="756" alt="Screen Shot 2021-06-21 at 12 25 38" src="https://user-images.githubusercontent.com/5055789/122796431-c1011b00-d294-11eb-9ac9-751e8ffc267d.png">
<img width="736" alt="Screen Shot 2021-06-21 at 12 25 48" src="https://user-images.githubusercontent.com/5055789/122796443-c3fc0b80-d294-11eb-974c-02e519fb52be.png">
That is not the case for actions create, so it's probably a bug..

Reply:
Removed in https://github.com/auth0/auth0-cli/pull/316/commits/ae517fa60da115323ded70d0c35765b81a57348f, as it does not fail anymore.